sdks/python/apache_beam/io/gcp/bigquery_file_loads.py

```diff
@@ -857,6 +972,8 @@ def _load_data(
          of the load jobs would fail but not other. If any of them fails, then
          copy jobs are not triggered.
     """
+    singleton_pc = p | "ImpulseLoadData" >> beam.Create([None])
```

Comment:
Would this break the improvement made in #13601 ? I can't tell when streaming pipelines can or cannot use FILE_LOADS.

Reply:
It'll be best to have separate singletons for each of these paths. The issue was because by having one single PCollection reused in all these paths was concentrating all of these various paths into a single stage with ~10 side inputs, and this complicates firing of triggers for the stage, because we need to wait for ~10 stages to advance their watermarks to start the 'singleton' stage.

If we have multiple singleton pcollections, then we have multiple stages with only one side input each.